### PR TITLE
Fixing Split-Concat reverse input channel

### PR DESCRIPTION
**Pull request by iimironov**
Ticket: 59156

The is copy of code from MR https://github.com/openvinotoolkit/openvino/pull/6506 for release branch.

Pattern Split->Concat was added before Pad op.
As a part of optimizations ReverseChannels operation was replacing to pattern Split->Сoncat. Earlier if the pattern inserted before Pad layer we fused it to Pad. But it was wrong and the optimization was removed. So the root cause is that before Pad we insert  Split->Сoncat pattern and it leads to regression for some topologies.

Screenshot (18)

The fix moves ReverseChannels after Pad layer. It allows optimize reverse input channels propagation and avoid work around with insert Split->Сoncat subgraph.

* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR N/A
* [x]  Transformation preserves original framework node names N/A: a special case, when we should rename the nodes

Validation:
* [x]  Unit tests
* [x]  Framework operation tests N/A
* [x]  Transformation tests N/A
* [x]  Model Optimizer IR Reader check N/A

Documentation:
* [x]  Supported frameworks operations list N/A: no new operations were enabled
* [x]  Guide on how to convert the **public** model N/A no new models were enabled
* [x]  User guide update N/A

### CODE REVIEW

**Review by yekruglov — APPROVED** [review]

**Review by rkazants — CHANGES_REQUESTED** [review]
@iimironov, please fix all typos (in title and description).

File: model-optimizer/extensions/back/ReverseInputChannels.py [hunk 1]
```diff
@@ -94,31 +94,56 @@ class ReverseChannelsPropagationDown(BackReplacementPattern):
...
         'ShapeOf': lambda node, rc: ReverseChannelsPropagationDown.pass_rc_through_shape(node, rc),
+
+        'Pad': lambda node, rc: ReverseChannelsPropagationDown.pass_rc_through(node, rc),
     }
 
+    @staticmethod
+    def pass_rc_through(node: Node, reverse_channels: Node):
+        r"""
+        BEFORE                          AFTER
+          previous_op
```
Comment (rkazants) [inline model-optimizer/extensions/back/ReverseInputChannels.py:105]: please add blank line between them.

File: model-optimizer/extensions/back/ReverseInputChannels.py [hunk 2]
```diff
@@ -170,15 +195,13 @@ def pass_rc_through_conv(node, reverse_channels):
...
-
           previous_op                                       previous_op'
               |                                                 |
         ReverseChannels  previous_op'    previous_op     ReverseChannels
                      \     /                        \     /
                      Eltwise                        Eltwise
                                                       |
                                                 ReverseChannels
-
         returns boolean value whatever we should continue propagating current ReverseChannels operation down or not
```
Comment (rkazants) [inline model-optimizer/extensions/back/ReverseInputChannels.py:205]: please do not edit these comments.

File: model-optimizer/extensions/back/ReverseInputChannels.py [hunk 3]
```diff
@@ -265,13 +288,41 @@ class ReverseChannelsPropagationUp(BackReplacementPattern):
...
         'Convert': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through_eltwise(node, rc),
+
+        'Pad': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through(node, rc),
     }
 
+    @staticmethod
+    def lift_up_through(node: Node, reverse_channels: Node):
+        r"""
+        BEFORE                       AFTER
+                                     previous_op
```
Comment (rkazants) [inline model-optimizer/extensions/back/ReverseInputChannels.py:299]: the same as above

**Review by rkazants — APPROVED** [review]

**Pull request merged**